datalad/interface/common_opts.py

```diff
@@ -139,8 +139,13 @@
     EnsureChoice(None, True, False, 'auto', 'ephemeral') | \
     EnsureStrPrefix('shared-'),
     metavar='auto|ephemeral|shared-...',
-    doc="""set up the dataset in a potentially unsafe way for performance,
-    or access reasons -- use with care, any dataset is marked as 'untrusted'.
+    doc="""Use this parameter with a command that clones a dataset
+    [PY: (`clone(<source>, reckless=True)`) PY]
+    [CMD: ('clone <source> --reckless') CMD]
+    or subdataset [PY: (`dl.get(<subds>, reckless=True)`) PY]
+    [CMD: ('datalad get <subds> --reckless') CMD]
+    to set up the dataset in a potentially unsafe way for performance, or
+    access reasons. Use with care, any dataset is marked as 'untrusted'.
```

Comment:
Yeah, its difficult. The place where confusion arises is in case of a ``get`` of anything that is not a subdataset (the most common usage of ``get``).
The explicit mentioning of the type of content (``dataset``, ``subdataset``) this flag applies to is a big improvement. I like this part of your suggestion. Maybe this additional change to your suggestion makes it clear that the flag should be used at the time of "obtaining ds/subds" instead of a post-hoc setup:
```suggestion
    doc="""Obtain a dataset or subdatset and set it up in a potentially 
    unsafe way for performance, or access reasons. 
    Use with care, any dataset is marked as 'untrusted'.
```
I'm always in favor of being more explicit.